pertpy/tools/_distances/_distances.py

```diff
@@ -323,3 +351,83 @@ def __call__(self, X: np.ndarray, Y: np.ndarray, **kwargs) -> float:
 
     def from_precomputed(self, P: np.ndarray, idx: np.ndarray, **kwargs) -> float:
         return P[idx, :][:, ~idx].mean()
+
+
+class PseudobulkPearsonDistance(AbstractDistance):
+    """Pearson distance between pseudobulk vectors."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.accepts_precomputed = False
+
+    def __call__(self, X: np.ndarray, Y: np.ndarray, **kwargs) -> float:
+        return 1 - pearsonr(X.mean(axis=0), Y.mean(axis=0))[0]
+
+    def from_precomputed(self, P: np.ndarray, idx: np.ndarray, **kwargs) -> float:
+        raise NotImplementedError("PseudobulkPearsonDistance cannot be called on a pairwise distance matrix.")
+
+
+class PseudobulkSpearmanDistance(AbstractDistance):
+    """Spearman distance between pseudobulk vectors."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.accepts_precomputed = False
+
+    def __call__(self, X: np.ndarray, Y: np.ndarray, **kwargs) -> float:
+        return 1 - spearmanr(X.mean(axis=0), Y.mean(axis=0))[0]
+
+    def from_precomputed(self, P: np.ndarray, idx: np.ndarray, **kwargs) -> float:
+        raise NotImplementedError("PseudobulkSpearmanDistance cannot be called on a pairwise distance matrix.")
+
+
+class PseudobulkCosineDistance(AbstractDistance):
+    """Cosine distance between pseudobulk vectors."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.accepts_precomputed = False
+
+    def __call__(self, X: np.ndarray, Y: np.ndarray, **kwargs) -> float:
+        return cosine(X.mean(axis=0), Y.mean(axis=0))
+
+    def from_precomputed(self, P: np.ndarray, idx: np.ndarray, **kwargs) -> float:
+        raise NotImplementedError("PseudobulkCosineDistance cannot be called on a pairwise distance matrix.")
+
+
+class PseudobulkR2Score(AbstractDistance):
+    """Coefficient of determination across genes between pseudobulk vectors."""
+
+    # NOTE: This is not a distance metric but a similarity metric.
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.accepts_precomputed = False
+
+    def __call__(self, X: np.ndarray, Y: np.ndarray, **kwargs) -> float:
+        return 1 - r2_score(X.mean(axis=0), Y.mean(axis=0))
```

Comment:
To make it a distance function as we are implementing it in the distance module not a similarity module. 
like pearson distance instead of pearson correlation. I know there is nothing called 1-r2 but I was not sure what to do. I left it this way to get your opinion...

Reply:
Ahh, I see what you mean. That's a great thought! Let's leave it like this for now.